libvast/vast/detail/type_traits.hpp

```diff
@@ -156,4 +156,13 @@ using remove_optional_t = typename remove_optional<T>::type;
 template <auto... Values>
 constexpr auto sum = (0 + ... + Values);
 
+template <class T, template <class...> class TList, class... Ts>
+constexpr auto contains_type(TList<Ts...>) {
+  return std::bool_constant<is_any_v<T, Ts...>>{};
+}
```

Comment:
It's a bit weird that this is not a struct, which makes it hard to use with other meta-programming facilities. I don't have a particularly hard opinion on this, but if you have an idea how to rewrite this easily as a struct that inherits from `std::bool_constant` than I'd be more than happy to see that change applied.

Reply:
Can you elaborate on the part about hard to use with other meta programming facilities? I just added an alias template `contains_type_t` if you want a type flavor of it (but I don't think it will ever get used in that way, in practice).

I'm not sure why you think it's weird that it's not a struct. With template templates, in deducing the types in the class template, you need to instantiate the type, which you could do in a constructor perhaps. We could maybe do something with a templated constructor and initialize a base class with the value based on `is_any_v` containment. That's some gymnastics for no benefit though IMO.

Reply:
Take for example `disjunction<A, B, C>::value`, which is logically equivalent to `A::value || B::value || C::value`. I cannot use this type trait as part of such a conjunction, disjunction, or negation.

I'm fine with this as-is, except for one minor naming request.

For a given type trait `A`, I expected `A_t` to be `typename A::type` and `A_v` to be `A::value`. The naming of the helper function should not be `A`.

Reply:
Will do should we ever need this. Thanks for going through this elongated review with me. 👍

Reply:
@dominiklohmann https://wandbox.org/permlink/HpHyHTiP4NaN5aiq is an example of what I think you're after.